recipes/pybind11/all/conanfile.py

```diff
@@ -16,83 +15,63 @@ class PyBind11Conan(ConanFile):
-        if Version(self.version) >= "2.6.0":
-            self.cpp_info.components["main"].set_property("cmake_module_file_name", "pybind11")
-            self.cpp_info.components["main"].names["cmake_find_package"] = "pybind11"
-            self.cpp_info.components["main"].builddirs = [cmake_base_path]
-            cmake_file = os.path.join(cmake_base_path, "pybind11Common.cmake")
-            self.cpp_info.set_property("cmake_build_modules", [cmake_file])
-            for generator in ["cmake_find_package", "cmake_find_package_multi"]:
-                self.cpp_info.components["main"].build_modules[generator].append(cmake_file)
-            self.cpp_info.components["headers"].includedirs = [os.path.join("include", "pybind11")]
-            self.cpp_info.components["headers"].requires = ["main"]
-            self.cpp_info.components["embed"].requires = ["main"]
-            self.cpp_info.components["module"].requires = ["main"]
-            self.cpp_info.components["python_link_helper"].requires = ["main"]
-            self.cpp_info.components["windows_extras"].requires = ["main"]
-            self.cpp_info.components["lto"].requires = ["main"]
-            self.cpp_info.components["thin_lto"].requires = ["main"]
-            self.cpp_info.components["opt_size"].requires = ["main"]
-            self.cpp_info.components["python2_no_register"].requires = ["main"]
-        else:
-            self.cpp_info.includedirs.append(os.path.join(
-                self.package_folder, "include", "pybind11"))
-
-            self.cpp_info.builddirs = [cmake_base_path]
-
-            cmake_files = [os.path.join(cmake_base_path, "FindPythonLibsNew.cmake"),
-                           os.path.join(cmake_base_path, "pybind11Tools.cmake")]
-            self.cpp_info.set_property("cmake_build_modules", cmake_files)
-            for generator in ["cmake", "cmake_multi", "cmake_find_package", "cmake_find_package_multi"]:
-                self.cpp_info.build_modules[generator] = cmake_files
+        self.cpp_info.set_property("cmake_target_name", "pybind11_all_do_not_use")
+        self.cpp_info.components["headers"].includedirs = ["include"]
+        self.cpp_info.components["pybind11_"].set_property("cmake_target_name", "pybind11::pybind11")
+        self.cpp_info.components["pybind11_"].set_property("cmake_module_file_name", "pybind11")
+        self.cpp_info.components["pybind11_"].names["cmake_find_package"] = "pybind11"
+        self.cpp_info.components["pybind11_"].builddirs = [cmake_base_path]
+        self.cpp_info.components["pybind11_"].requires = ["headers"]
+        cmake_file = os.path.join(cmake_base_path, "pybind11Common.cmake")
+        self.cpp_info.set_property("cmake_build_modules", [cmake_file])
+        for generator in ["cmake_find_package", "cmake_find_package_multi"]:
+            self.cpp_info.components["_pybind11"].build_modules[generator].append(cmake_file)
+        self.cpp_info.components["embed"].requires = ["_pybind11"]
```

Comment:
Above the component is `pybind11_` which does it change?

Reply:
typo